ee/clickhouse/queries/retention/retention_event_query.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Dict, Literal, Tuple
+
+from ee.clickhouse.models.action import format_action_filter
+from ee.clickhouse.queries.event_query import ClickhouseEventQuery
+from ee.clickhouse.queries.util import get_trunc_func_ch
+from posthog.constants import PAGEVIEW_EVENT, TREND_FILTER_TYPE_ACTIONS, TREND_FILTER_TYPE_EVENTS, TRENDS_LINEAR
+from posthog.models import Entity
+from posthog.models.action import Action
+from posthog.models.filters.retention_filter import RetentionFilter
+
+
+class RetentionEventsQuery(ClickhouseEventQuery):
+    _filter: RetentionFilter
+    _event_query_type: Literal["returning", "target", "target_first_time"]
+    _trunc_func: str
+
+    def __init__(self, event_query_type: Literal["returning", "target", "target_first_time"], *args, **kwargs):
+        self._event_query_type = event_query_type
+        super().__init__(*args, **kwargs)
+
+        self._trunc_func = get_trunc_func_ch(self._filter.period)
+
+    def get_query(self) -> Tuple[str, Dict[str, Any]]:
+        _fields = (
```

Comment:
thoughts on following the new format here?:

https://github.com/PostHog/posthog/blob/master/ee/clickhouse/queries/paths/path_event_query.py#L41

I'm yet to update trends, but funnels & paths follow this now. Makes it easier to handle commas + empty strings